pkg/disruption/backend/sampler/remote.go

```diff
@@ -0,0 +1,501 @@
+		events = append(events, nodeEvents...)
+	}
+	if len(errs) > 0 {
+		framework.Logf("found errors fetching in-cluster data: %v", errs)
+	}
+	artifactPath := filepath.Join(os.Getenv("ARTIFACT_DIR"), inClusterEventsFile)
+	return monitorserialization.EventsToFile(artifactPath, events)
+}
+
+func StartInClusterMonitors(ctx context.Context, config *rest.Config) error {
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	configClient, err := configclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	internalAPI, err := url.Parse(infra.Status.APIServerURL)
+	if err != nil {
+		return err
+	}
+	apiIntHost := strings.Replace(internalAPI.Hostname(), "api.", "api-int.", 1)
+
+	projectClient, err := projectclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	err = callProject(ctx, projectClient, true)
+	if err != nil {
+		return err
+	}
+	err = callServiceAccount(ctx, kubeClient, true)
+	if err != nil {
+		return err
+	}
+	err = callRBACClusterAdmin(ctx, kubeClient, true)
+	if err != nil {
+		return err
+	}
+	err = callRBACHostaccess(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	return callMonitorDaemonset(ctx, kubeClient, apiIntHost, true)
+}
+
+func deleteTestBed(ctx context.Context, projectClient *projectclient.Clientset, kubeClient *kubernetes.Clientset) error {
+	// Stop daemonsets first so that test stop before serviceaccount is removed
+	// and permission issues from apiserver are not recorded as disruption
+	err := callMonitorDaemonset(ctx, kubeClient, "", false)
+	if err != nil {
+		return err
+	}
+
+	err = callRBACClusterAdmin(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	err = callRBACHostaccess(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	err = callServiceAccount(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	return callProject(ctx, projectClient, false)
+}
+
+func createDaemonset(ctx context.Context, clientset *kubernetes.Clientset, create bool, obj *appsv1.DaemonSet) error {
+	client := clientset.AppsV1().DaemonSets(namespace)
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		if err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 5*time.Minute, true, func(context.Context) (bool, error) {
+			ds, err := client.Get(context.Background(), obj.Name, metav1.GetOptions{})
+			if err != nil {
+				framework.Logf("error getting daemonsets %v", err)
+				return false, nil
+			}
+			return ds.Status.NumberReady > 0, nil
+		}); err != nil {
+			return fmt.Errorf("daemonset %s didn't roll out: %v", obj.Name, err)
+		}
+
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
```

Comment:
don't bother deleting individual things.  if something fails, nuke the entire namespace.